pkg/reconciler/instances/ory/ory.go

```diff
@@ -0,0 +1,27 @@
+package ory
+
+import (
+	"github.com/kyma-incubator/reconciler/pkg/logger"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/service"
+)
+
+const ReconcilerName = "ory"
+
+//nolint:gochecknoinits //usage of init() is intended to register reconciler-instances in centralized registry
+func init() {
+	log := logger.NewLogger(false)
+
+	log.Debugf("Initializing component reconciler '%s'", ReconcilerName)
+	reconciler, err := service.NewComponentReconciler(ReconcilerName)
+	if err != nil {
+		log.Fatalf("Could not create '%s' component reconciler: %s", ReconcilerName, err)
+	}
+
+	reconciler.
+		WithPreReconcileAction(&ReconcileAction{
+			step: "pre-install",
```

Comment:
I would recommend to create dedicated Actions as they do different things (separation of concerns).

Things which are shared between both actions can be handled via Mixins:

     type oryAction struct {
         ...shared resources / fcts end up in this struct...
     }

    type preAction struct {
       *oryAction
     }
     func (a *preAction) Run(...) {}

    type postAction struct {
       *oryAction
     }
     func (a *postAction) Run(...) {}

Reply:
good idea